apps/07jquery-tic-tac-toe/script.js

```diff
@@ -1,5 +1,57 @@
 'use strict';
-
+var playerTurn = "X";
 $(document).on('ready', function() {
-  // Put app logic in here
+    // Put app logic in here
+    var $datacells = $('[data-cell]')
+    $datacells.click(function() {
+        $(this).text(playerTurn);
+        checkForWin();
+        if (playerTurn === 'X') {
+            playerTurn = 'O';
+        } else {
+            playerTurn = 'X';
+        }
+        //Add a button with an id="clear" that will not only clear the board,
+        //but reset the player to player 'X'
+
+
+    });
+$('#clear').click(function() {
+$('[data-cell]').text("");
+$('#announce-winner').text("");
+playerTurn = 'X';
 });
+});
+
+
+function horizontalWin() {
+    // Your code here
+    if ($('[data-cell = "0"]').text() === playerTurn &&  $('[data-cell = "1"]').text() === playerTurn &&  $('[data-cell = "2"]').text() === playerTurn ||  $('[data-cell = "3"]').text() === playerTurn &&  $('[data-cell = "4"]').text() === playerTurn &&  $('[data-cell = "5"]').text() === playerTurn ||  $('[data-cell = "6"]').text() === playerTurn && $('[data-cell = "7"]').text() === playerTurn && $('[data-cell = "8"]').text() === playerTurn) {
+      return true;
+    }
+    return false;
+}
+
+function verticalWin() {
+    // Your code here
+if ($('[data-cell = "0"]').text() === playerTurn &&  $('[data-cell = "3"]').text() === playerTurn &&  $('[data-cell = "6"]').text() === playerTurn ||  $('[data-cell = "1"]').text() === playerTurn &&  $('[data-cell = "4"]').text() === playerTurn &&  $('[data-cell = "7"]').text() === playerTurn ||  $('[data-cell = "2"]').text() === playerTurn && $('[data-cell = "5"]').text() === playerTurn && $('[data-cell = "8"]').text() === playerTurn) {
+  return true;
+}
+     return false;
+}
+
+function diagonalWin() {
+    // Your code here
+  if($('[data-cell = "0"]').text() === playerTurn &&  $('[data-cell = "4"]').text() === playerTurn &&  $('[data-cell = "8"]').text() === playerTurn ||  $('[data-cell = "2"]').text() === playerTurn &&  $('[data-cell = "4"]').text() === playerTurn &&  $('[data-cell = "6"]').text() === playerTurn) {
```

Comment:
Your logic checks out. If you want to really consolidate your code, you can use a `for` loop to run through your jquery logic using string concatenation. 

example: ```$('[data-cell="'+ [i][0] + '"]').text() === playerTurn```